src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6435,7 +6458,34 @@ private FlowAnalysisAnnotations GetParameterAnnotations(ParameterSymbol paramete
         {
             // Annotations are ignored when binding an attribute to avoid cycles. (Members used
             // in attributes are error scenarios, so missing warnings should not be important.)
-            return IsAnalyzingAttribute ? FlowAnalysisAnnotations.None : parameter.FlowAnalysisAnnotations;
+            if (IsAnalyzingAttribute)
+                return FlowAnalysisAnnotations.None;
+
+            // Conditional annotations are ignored on parameters of non-boolean members.
+            if (parameter.ContainingSymbol.GetTypeOrReturnType().Type.SpecialType != SpecialType.System_Boolean)
```

Comment:
>if [](http://example.com/codeflow?start=12&length=2)

What drove this change?

Reply:
Scenarios like `NotNullWhenTrue_ReturningObject` and the other related tests I added `NotNullWhen***_Returning***`. They would also hit the newly added assertion, leaving a split state after analyzing an invocation returning `object` or some other types.